llvm/lib/Target/X86/X86MachineFunctionInfo.h

```diff
@@ -174,6 +175,9 @@ class X86MachineFunctionInfo : public MachineFunctionInfo {
   bool FPClobberedByCall = false;
   bool BPClobberedByCall = false;
 
+  // A set of instructions clobber FP generated from longjmp.
+  SmallSet<MachineInstr *, 2> LongJmpMIClobberFP;
```

Comment:
I don't think it is safe to store MIR pointers in side tables from pass-to-pass. It is too likely that a MachineInstr will be cloned and replaced, invalidating these pointers, leading to an ABA-style use-after-free.

It also occurs to me that we may need to make sure that X86MachineFunctionInfo is fully serialized through MIR tests. I would have to look further into that.

Reply:
Longjmp modifies fp and sp, these instructions can be treated as destroying current stack frame and setting up stack frame for the jump target.

How about set FrameDestroy flag for these longjmp instructions, then spillFPBP can ignore these instructions.

Reply:
I like that approach, although there is some risk that the FrameDestroy flag is used to detect epilogue code. Longjmp instructions are effectively epilogue code: it tears down the frame and replaces it.